test/suites/basic.sh

```diff
@@ -1173,3 +1173,149 @@ EOF
 
   services_validator
 }
+
+test_remove_cluster_member() {
+  unset_interactive_vars
+
+  microcloud_internal_net_addr="$(ip_config_to_netaddr lxdbr0)"
+
+  # Set the default config for interactive setup.
+  export LOOKUP_IFACE="enp5s0"
+  export LIMIT_SUBNET="yes"
+  export EXPECT_PEERS=2
+  export SETUP_ZFS="yes"
+  export ZFS_FILTER="lxd_disk1"
+  export ZFS_WIPE="yes"
+  export SETUP_CEPH="yes"
+  export SETUP_CEPHFS="yes"
+  export CEPH_FILTER="lxd_disk2"
+  export CEPH_WIPE="yes"
+  export CEPH_CLUSTER_NETWORK="${microcloud_internal_net_addr}"
+  export CEPH_ENCRYPT="no"
+  export SETUP_OVN="yes"
+  export OVN_FILTER="enp6s0"
+  export IPV4_SUBNET="10.1.123.1/24"
+  export IPV4_START="10.1.123.100"
+  export IPV4_END="10.1.123.254"
+  export DNS_ADDRESSES="10.1.123.1,8.8.8.8"
+  export IPV6_SUBNET="fd42:1:1234:1234::1/64"
+
+  reset_systems 3 3 3
+  echo "Fail to remove member from MicroCeph and LXD until OSDs are removed"
```

Comment:
Where are the OSDs being removed? I am not really following why the first invocation (without force) of `microcloud remove micro02` is failing or "secured" with `|| true`.
Shouldn't the removal succeed?

Reply:
Ah ok so MicroCeph is the limiting factor here?

Reply:
That's a MicroCeph restriction that does not allow the cluster remove API to proceed if there are OSDs remaining. We call `DeleteClusterMember` for MicroCeph, which triggers its `PreRemove` hook where the restrictions are handled.